2024-RIKEN-AWS/JupyterNotebook/docker/Dockerfile.spawn

```diff
@@ -96,11 +124,14 @@ COPY ./docker/entrypoint.sh /entrypoint.sh
 
 # This is for a local start
 COPY ./docker/start.sh /start.sh
-CMD ["flux", "start", "--test-size=4", "jupyter", "lab"]
 
-# This won't be available in K8s, but will be for a single container build
-COPY ./tutorial /home/jovyan/flux-tutorial-2024
+RUN mkdir -p $HOME/.local/share && \
+    chmod 777 $HOME/.local/share
```

Comment:
What is going to go here?

Reply:
The diff seems to be a little confused there. Two things are happening:
1. The `COPY` directive was moved towards the top of the file
2. We added a `chmod` to `/home/jovyan/.local/share` because we were getting permission denied errors on that directory

Reply:
I think this is happening because you are running all of these as `USER root`. Running the add user command does not switch the user, it just adds them. So the commands to install (with sudo) if I'm reading this right are being run by root, and the COPY directives would have root permissions too. If you want the jovyan user to have ownership of that space I would also COPY, etc. in the context of `USER jovyan`. Generally the pattern to follow is:

1. Do all system installs as root
2. Change to the user that should own files
3. Then copy them in

Reply:
That makes sense. To be honest, I'm not entirely sure why it's complaining about `$HOME/.local/share` anymore. To my knowledge, nothing is being copied into there.